x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields_modal.tsx

```diff
@@ -0,0 +1,286 @@
+    defaultMessage: '{ruleName} was saved',
+  });
+
+const ADD_CUSTOM_FIELD_LABEL = i18n.translate(
+  'xpack.securitySolution.flyout.right.investigation.highlightedFields.modalAddCustomFieldLabel',
+  { defaultMessage: 'Add custom' }
+);
+
+const SELECT_PLACEHOLDER = i18n.translate(
+  'xpack.securitySolution.flyout.right.investigation.highlightedFields.modalSelectPlaceholder',
+  { defaultMessage: 'Select or search for options' }
+);
+
+const CommonUseField = getUseField({ component: Field });
+
+interface InvestigationFieldsFormData {
+  customHighlightedFields: string[];
+}
+
+const schema: FormSchema<InvestigationFieldsFormData> = {
+  customHighlightedFields: {
+    fieldsToValidateOnChange: ['customHighlightedFields'],
+    type: FIELD_TYPES.COMBO_BOX,
+    validations: [{ validator: fieldValidators.emptyField('error') }],
+  },
+};
+
+const formConfig = {
+  ...schema.customHighlightedFields,
+  label: ADD_CUSTOM_FIELD_LABEL,
+};
+
+interface HighlightedFieldsModalProps {
+  /**
+   * The data formatted for field browser
+   */
+  dataFormattedForFieldBrowser: TimelineEventsDetailsItem[];
+  /**
+   * The rule
+   */
+  rule: RuleResponse;
+  /**
+   * The custom highlighted fields
+   */
+  customHighlightedFields: string[];
+  /**
+   * The function to set the edit loading state
+   */
+  setIsEditLoading: (isEditLoading: boolean) => void;
+  /**
+   * The function to set the modal visible state
+   */
+  setIsModalVisible: (isModalVisible: boolean) => void;
+}
+
+/**
+ * Modal for editing the highlighted fields of a rule.
+ */
+export const HighlightedFieldsModal: FC<HighlightedFieldsModalProps> = ({
+  rule,
+  customHighlightedFields,
+  dataFormattedForFieldBrowser,
+  setIsEditLoading,
+  setIsModalVisible,
+}) => {
+  const defaultIndexPattern = useDefaultIndexPattern();
+  const { dataSourceType, index, dataViewId } = useMemo(() => getDefineStepsData(rule), [rule]);
+  const { indexPattern: dataView } = useRuleIndexPattern({
+    dataSourceType,
+    index: index.length > 0 ? index : defaultIndexPattern, // fallback to default index pattern if rule has no index patterns
+    dataViewId,
+  });
+  const fieldOptions = useMemo(() => dataView.fields, [dataView]);
+
+  const { addSuccess } = useAppToasts();
+  const { euiTheme } = useEuiTheme();
+  const { mutateAsync: updateRule } = useUpdateRule();
+  const modalTitleId = useGeneratedHtmlId();
+
+  const defaultFields = useHighlightedFields({
+    dataFormattedForFieldBrowser,
+    investigationFields: customHighlightedFields,
+    type: 'default',
+  });
+  const defaultFieldsArray = useMemo(() => Object.keys(defaultFields), [defaultFields]);
+
+  const customFields = useMemo(
+    () => customHighlightedFields.map((field: string) => ({ label: field })),
+    [customHighlightedFields]
+  );
+
+  const [selectedOptions, setSelectedOptions] = useState(customFields);
+
+  const options = useMemo(
+    () => fieldOptions.map((field) => ({ label: field.name })),
+    [fieldOptions]
```

Comment:
Sorry for not catching this in my first review, but what do you think about filtering out the values that are already in the `defaultFieldsArray`?

To avoid having something like this:

https://github.com/user-attachments/assets/c6ac9a16-b85d-475c-827a-dc29acc27d76

It's not that big of a deal, but could be a nice touch, wdyt?